### PR TITLE
use shared context for active storage setup

### PR DESCRIPTION
**Pull request by jsugarman**

### CODE REVIEW

File: spec/models/document_spec.rb [hunk 1]
```diff
@@ -321,158 +344,80 @@ def make_empty_temp_file
...
-
-        service = ActiveStorage::Service::DiskService.new(root: '/root/')
-        allow(ActiveStorage::Blob).to receive(:service).and_return(service)
+        stub_const 'PAPERCLIP_STORAGE_PATH', 'public/assets/test/images/:id_partition/:filename'
       end
 
-      it 'has the path for disk storage' do
-        expect(document.document.path).to eq '/root/te/st/test_key_original'
+      context 'without an Active Storage attachment' do
+        it { expect(document.document.path).to eq "public/assets/test/images/#{id_partition}/#{filename}" }
```
Comment (jrmhaig) [inline spec/models/document_spec.rb]: As you have set a `subject` did you want to do:

```suggestion
        it { is_expected.to eq "public/assets/test/images/#{id_partition}/#{filename}" }
```

File: spec/models/document_spec.rb [hunk 2]
```diff
@@ -321,158 +344,80 @@ def make_empty_temp_file
...
-        service = ActiveStorage::Service::S3Service.new(bucket: 'bucket')
-        allow(ActiveStorage::Blob).to receive(:service).and_return(service)
-      end
+        include_context 'add active storage record assets' do
+          let(:service) { ActiveStorage::Service::DiskService.new(root: '/root/') }
+        end
 
-      it 'has the path for disk storage' do
-        expect(document.document.path).to eq 'test_key_original'
+        it { expect(document.document.path).to eq '/root/te/st/test_key_original' }
```
Comment (jrmhaig) [inline spec/models/document_spec.rb]: Likewise,

```suggestion
        it { is_expected.to eq '/root/te/st/test_key_original' }
```

File: spec/models/document_spec.rb [hunk 3]
```diff
@@ -321,158 +344,80 @@ def make_empty_temp_file
...
-      stub_const 'PAPERCLIP_STORAGE_PATH', 'public/assets/test/images/:id_partition/:filename'
-    end
+        include_context 'add active storage record assets' do
+          let(:service) { ActiveStorage::Service::S3Service.new(bucket: 'bucket') }
+        end
 
-    context 'without an Active Storage attachment' do
-      it 'has a path based on the filename' do
-        expect(document.converted_preview_document.path).to eq "public/assets/test/images/#{id_partition}/#{filename}"
+        it { expect(document.document.path).to eq 'test_key_original' }
```
Comment (jrmhaig) [inline spec/models/document_spec.rb]: Likewise,

```suggestion
        it { is_expected.to eq 'test_key_original' }
```

**Review by jrmhaig — APPROVED** [review]
Thanks. Looks OK. A small suggestion that you can change if you want but not necessary.

File: spec/models/document_spec.rb (hunk 1 above)
Comment (jsugarman) [reply spec/models/document_spec.rb]: yes, i meant to but was in a rush (school run 😄 )

**Pull request merged**